packages/source-fetcher/lib/sourcify.ts

```diff
@@ -0,0 +1,136 @@
+import debugModule from "debug";
+const debug = debugModule("source-fetcher:sourcify");
+
+import { Fetcher, FetcherConstructor } from "./types";
+import * as Types from "./types";
+import { removeLibraries } from "./common";
+import request from "request-promise-native";
+
+//this looks awkward but the TS docs actually suggest this :P
+const SourcifyFetcher: FetcherConstructor = class SourcifyFetcher
+  implements Fetcher {
+  get fetcherName(): string {
```

Comment:
I feel like I'm missing something here, but why is there a `get fetcherName()` _and_ a `static get fetcherName()`? Is one of these in here accidentally?

Reply:
No, I want it to be available both statically and per-instance.  Strictly speaking this is a little redundant, as you could always use `.constructor`, but that seems to run into Typescript problems? :-/  (OK, to be sure, I'm only using this from JS, but ideally it should be usable from TS as well.)

You might want to see the [code](https://github.com/trufflesuite/truffle/blob/develop/packages/core/lib/debug/external.js) where this is actually used -- as you can see, we do indeed use it both on the class (for determining which ones to use and in what order, and for getting the config options for each) and on the instance (for error reporting and for assigning compilation IDs).